numba/tests/test_dispatcher.py

```diff
@@ -1306,5 +1306,63 @@ def cache_file_collision_tester(q, tempdir, modname_bar1, modname_bar2):
     q.put(r2)
 
 
+class TestDispatcherFunctionBoundaries(TestCase):
+    def test_pass_dispatcher_as_arg(self):
```

Comment:
Perhaps add a test for passing dispatchers in containers as that seems to be a commonly asked for feature?

Reply:
I don't think that will work in any useful way.  Given how each dispatcher is itself a unique type and we don't have polymorphic function type, we can only type `fn = many_functions[i]` iff 

* `i` is a constant and `many_functions` is a tuple of dispatchers; such that we can deduce which tuple-element the code is referring to at compile-time.
* or, `many_functions` contain exactly one dispatcher type but can have many copies in it. 

Neither way can make useful code.

At the very least, I can make a test for the two silly cases.

Reply:
Ah yes, the index would have to be const to allow type resolution, these cases are indeed silly :)

Reply:
I have added the tests.